examples/schema.rs

```diff
@@ -0,0 +1,137 @@
+    let command_args = command_args.unwrap();
+
+    match command_name {
+        "load" => load(command_args)?,
+        "validate" => validate(command_args)?,
+        _ => eprintln!("command name: {} not found", command_name),
+    }
+
+    Ok(())
+}
+
+fn load(command_args: &ArgMatches) -> IonSchemaResult<()> {
+    // Extract the user provided authorities
+    let authorities: Vec<_> = command_args.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = command_args.value_of("schema").unwrap();
+
+    // Set up authorities vector
+    let mut document_authorities: Vec<Box<dyn DocumentAuthority>> = vec![];
+
+    for authority in authorities {
+        document_authorities.push(Box::new(FileSystemDocumentAuthority::new(Path::new(
+            authority,
+        ))))
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    if schema.is_ok() {
+        eprintln!("Schema: {:?} was successfully loaded", schema.unwrap().id());
+    } else {
+        eprintln!("{:?}", schema.unwrap_err());
+    }
+
+    Ok(())
+}
+
+fn validate(command_args: &ArgMatches) -> IonSchemaResult<()> {
+    // Extract the user provided authorities
+    let authorities: Vec<_> = command_args.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = command_args.value_of("schema").unwrap();
+
+    // Extract Ion value provided by user
+    let input_file = command_args.value_of("input").unwrap();
+    let value = fs::read(input_file).expect("Can not load given ion file");
+    let owned_elements = element_reader()
+        .read_all(&value)
+        .expect("parsing failed unexpectedly");
+
+    // Set up authorities vector
+    let mut document_authorities: Vec<Box<dyn DocumentAuthority>> = vec![];
+
+    for authority in authorities {
+        document_authorities.push(Box::new(FileSystemDocumentAuthority::new(Path::new(
+            authority,
+        ))))
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    // get the type defined within the schema file
+    // there will only be a single type defined within schema
+    let type_ref = schema
+        .unwrap()
+        .get_types()
+        .next()
+        .expect("Loaded schema was empty");
+
+    // create a text writer to make the output
+    let mut output = vec![];
+    let mut writer = TextWriter::new(&mut output);
+
+    // validate owned_elements according to type_ref
+    for owned_element in owned_elements {
+        // create a validation report with validation result, value, schema and/or violation
```

Comment:
Can you provide a sample of what he validation report looks like for valid and invalid data? Either in this PR or in the documentation somewhere is fine.

Reply:
I have included validation report example in the PR description for this PR.
link: https://github.com/amzn/ion-schema-rust/pull/51#issue-1080479620